kroxylicious/src/main/java/io/kroxylicious/proxy/internal/filter/ApiVersionsFilter.java

```diff
@@ -23,12 +26,19 @@ public class ApiVersionsFilter implements ApiVersionsResponseFilter {
     private static final Logger LOGGER = LoggerFactory.getLogger(ApiVersionsFilter.class);
 
     private static void intersectApiVersions(String channel, ApiVersionsResponseData resp) {
+        Set<ApiVersionsResponseData.ApiVersion> toDelete = new HashSet<>();
         for (var key : resp.apiKeys()) {
             short apiId = key.apiKey();
             if (ApiKeys.hasId(apiId)) {
                 ApiKeys apiKey = ApiKeys.forId(apiId);
                 intersectApiVersion(channel, key, apiKey);
             }
+            else {
+                toDelete.add(key);
+            }
+        }
+        for (ApiVersionsResponseData.ApiVersion apiVersion : toDelete) {
+            resp.apiKeys().remove(apiVersion);
         }
```

Comment:
Nit on the set naming:
```suggestion
        Set<ApiVersionsResponseData.ApiVersion> unknownApis = new HashSet<>();
        for (var key : resp.apiKeys()) {
            short apiId = key.apiKey();
            if (ApiKeys.hasId(apiId)) {
                ApiKeys apiKey = ApiKeys.forId(apiId);
                intersectApiVersion(channel, key, apiKey);
            }
            else {
                unknownApis.add(key);
            }
        }
        resp.apiKeys().removeAll(unknownApis);
```